### DISCUSSION TITLE
Motion Sensor II (2844-222) - ON Only

### DISCUSSION THREAD

**Discussion by jb1228** [TD22057/insteon-mqtt#461] [Q&A] [answered]
Hello!

I have a motion sensor in my bathroom to turn on/off an Insteon Dimmer Switch. I wanted to change it to only send ON commands and use Home Assistant to turn off the lights (and fan On/Off Switch) depending on humidity and door sensors. 

I tried the following command to allow this; but even though the command is successful, the motion sensor continues to send an OFF command:

```
>>>insteon-mqtt config.yaml set-flags bathroom_motion on_only=1
Commanding motion_sensor device xx.xx.xx (bathroom_motion) cmd=set_flags
Motion xx.xx.xx extended operating flags: 00011010
Device set_flags complete
```
Is there a trick to setting this?

As an alternative option, is there a way to achieve this by manually editing the `scenes.yaml` file? 

This is my current entry:  
``` yaml
  - controllers:
      - bathroom_motion
    responders:
      - bathroom_lights:
          ramp_rate: 0.5
```
I was kind of hoping I could even use separate groups on the motion sensor.... one to send both ON and OFF commands to my controller and use another group to only send ON commands to the lights. Not sure if that is possible though.

I know I could just use Home Assistant to fully control the lights, but I would really prefer to have the delay that route adds. 

Thank you!

**Comment by krkeegan** [comment]
I believe the on_only command should work.  The response:
```
Motion xx.xx.xx extended operating flags: 00011010
```

Seems to show the correct bit (the second from the left) as set to 1.  Maybe try pulling the battery?  Maybe it just needs to reset?

As to your second question `one to send both ON and OFF commands to my controller and use another group to only send ON commands to the lights` no this is no possible in Insteon.

**Reply by jb1228** [reply]
Hello, 

Thank you for the suggestion. I tried to pull the battery, but it didn't work. 

I just so happen to have 4 of these motion sensors, so I factory reset all 4 and tried again with each one. 

Here are the versions printed on each:

- R3.1 3817
- R3.1 3017
- R3.3 4418
- R3.3 4418

Here are the commands I ran (all 4 had the same results):

```
>>>insteon-mqtt config.yaml awake motion_sensor_2
Commanding motion_sensor device xx.xx.xx (motion_sensor_2) cmd=awake
BatterySensor xx.xx.xx (motion_sensor_2) marked as awake
Complete


>>>insteon-mqtt config.yaml join motion_sensor_2
Commanding motion_sensor device xx.xx.xx (motion_sensor_2) cmd=join
Device xx.xx.xx engine version: i2c
Device joined.


>>>insteon-mqtt config.yaml pair motion_sensor_2
Commanding motion_sensor device xx.xx.xx (motion_sensor_2) cmd=pair
Device xx.xx.xx refresh cmd2 150
Device xx.xx.xx db out of date (got 25 vs 33), refreshing
Entry: 00ff: xx.xx.xx (modem)          grp:   1 type: RESP data: 0x03 0x00 0x00
Entry: 00f7: 00.00.00                  grp:   0 type: RESP data: 0x00 0x00 0x00 (UNUSED) (LAST)
xx.xx.xx database download complete
DeviceDb: (delta 25)
  00ff: xx.xx.xx (modem)          grp:   1 type: RESP data: 0x03 0x00 0x00
Unused:
  00f7: 00.00.00                  grp:   0 type: RESP data: 0x00 0x00 0x00 (UNUSED) (LAST)
Last:
  00f7: 00.00.00                  grp:   0 type: RESP data: 0x00 0x00 0x00 (UNUSED) (LAST)
GroupMap

Device paired


>>>insteon-mqtt config.yaml get-flags motion_sensor_2
Commanding motion_sensor device xx.xx.xx (motion_sensor_2) cmd=get_flags
Device xx.xx.xx operating flags: 10001111
Operation complete


>>>insteon-mqtt config.yaml set-flags motion_sensor_2 on_only=1
Commanding motion_sensor device xx.xx.xx (motion_sensor_2) cmd=set_flags
Motion xx.xx.xx extended operating flags: 00011110
Device set_flags complete


>>>insteon-mqtt config.yaml get-flags motion_sensor_2
Commanding motion_sensor device xx.xx.xx (motion_sensor_2) cmd=get_flags
Device xx.xx.xx operating flags: 10001111
Operation complete


>>>insteon-mqtt config.yaml set-flags motion_sensor_2 on_only=0
Commanding motion_sensor device xx.xx.xx (motion_sensor_2) cmd=set_flags
Motion xx.xx.xx extended operating flags: 00011110
Device set_flags complete


>>>insteon-mqtt config.yaml get-flags motion_sensor_2
Commanding motion_sensor device xx.xx.xx (motion_sensor_2) cmd=get_flags
Device xx.xx.xx operating flags: 10001101
Operation complete


>>>insteon-mqtt config.yaml set-flags motion_sensor_2 on_only=1
Commanding motion_sensor device xx.xx.xx (motion_sensor_2) cmd=set_flags
Motion xx.xx.xx extended operating flags: 00001100
Device set_flags complete


>>>insteon-mqtt config.yaml get-flags motion_sensor_2
Commanding motion_sensor device xx.xx.xx (motion_sensor_2) cmd=get_flags
Device xx.xx.xx operating flags: 10001111
Operation complete
```

I didn't do anything after that (like link them to lights or anything). I just watched their statuses in Home Assistant to see them go on and off. I also tried one of them using USB power the whole time instead of a battery, but that didn't seem to make a difference either. 

Any suggestions?

**Reply by krkeegan** [reply]
No suggestions.  It is clearly changing that second bit.  Not sure why it isn't working.  Does the instruction manual contain any instructions on how to set it manually to on_only?  I remember some versions had dip switches, but I don't think this version does.

**Reply by jb1228** [reply]
Unfortunately, no. It only has the one jumper to decrease sensitivity (which I did try) and the tamper switch. 

I appreciate your reply. I'll keep working on it. If I find anything, I'll make sure to post a reply, just in case others have the same issue. 

Thank you!

**Comment by jb1228** [comment, marked as answer]
Figured it out!

This command will successfully enable the `on_only` mode for this motion sensor: 

```
raw-command bathroom_motion 0x2E 0x00 --crc D14 --ext-req -d 0x00 0x06 0x20 0x0F 0x01 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00
```

Source: https://forums.indigodomo.com/viewtopic.php?p=183576#p183576

> Try using the Execute Raw INSTEON Command with the following settings:
> 
> Command 1 Byte: 0x2E
> Command 2 Byte: 0x00
> Send extended messages: checked
> D1 Byte: 0x00
> D2 Byte: 0x06
> D3 Byte: 0x20 for ON only, or 0x00 for ON and OFF
> D4 Byte: 0x0F
> D5 Byte: OFF timeout interval in 10 second increments (0x03 would be 30 seconds). This value is probably unused if you have D3 set to 0x20.
> D6 to D14 Bytes: 0x00 (for all of them)

**Reply by krkeegan** [reply]
Awesome, thank you for figuring this out.  I will convert this to an issue to be fixed.

**Reply by krkeegan** [reply]
Alright.  It turns out I was the one who reverse engineered this years ago.  I just forgot about it.

The problem is on the device `on_only` and `night_only` are actually set to `false|0` when enabled.  I was paying attention to the bit positions and not their values, the data is right there in my notes.  Anyways, I took a device down and confirmed all of this all over again today.  #464 will fix this when merged and a new release is issued. 

In the interim, your code worked because it set the `on_only` bit to false which enabled on_only.  However, it also set `led_on` to off and `night_only` to on.  You can fix this with the original code by doing the following:

`set-flags <device> led_on=1`
`set-flags <device> night_only=1`  << As noted above, this will actually set night_only off until the new code.
